data-prepper-plugins/geoip-prepper/src/main/java/com/amazon/dataprepper/plugins/prepper/geoip/provider/GeoIpProviderFactory.java

```diff
@@ -17,7 +18,9 @@ public GeoIpProvider createGeoIpProvider(final PluginSetting pluginSetting) {
         switch (provider) {
             case MaxMindGeolite2CityDatabase:
                 String dbPath = pluginSetting.getStringOrDefault(GeoIpPrepperConfig.DATABASE_PATH, null);
-                String[] desiredFields = (String[]) pluginSetting.getAttributeOrDefault(GeoIpPrepperConfig.DESIRED_FIELDS, DEFAULT_DESIRED_FIELDS);
+                @SuppressWarnings("unchecked")
+                final List<String> fields = (List<String>) pluginSetting.getAttributeFromSettings(GeoIpPrepperConfig.DESIRED_FIELDS);
```

Comment:
nit: if we have other use case to get String array from PluginSetting consider adding method `getStringArrayOrDefault(..) or getStringArray(..) `in [PluginSetting class](https://github.com/opendistro-for-elasticsearch/data-prepper/blob/88e0236d99440ccaa3fb77673b7505d800b265c1/data-prepper-api/src/main/java/com/amazon/dataprepper/model/configuration/PluginSetting.java#L5)